scripts/report_scripts/variantreport_p_sample.Rmd

```diff
@@ -525,59 +525,68 @@ Called SNVs:
-# add columns for all possible variants to the dataframe
-for (variant in all_variants) {
-  output_variant_plot[,variant] <- numeric()
-}
-meta_data <- c (samplename = sample_name,
-                dates = date,
-                location_name = location_name,
-                coordinates_lat = coordinates_lat,
-                coordinates_long = coordinates_long)
-
-output_variant_plot <- bind_rows(output_variant_plot, meta_data)
-
-# get rownumber for current sample
-sample_row <- which(grepl(sample_name, output_variant_plot$samplename))
-
-# write mutation frequency values to df
-for (i in all_variants){
-  if (i %in% df$name){
-    # check if variant already has a column
-    if (i %in% colnames(output_variant_plot)) {
-        output_variant_plot[sample_row,][i] <- df$value[df$name == i]
-        output_variant_plot <- output_variant_plot %>% mutate(others = 1-rowSums(across(all_of(all_variants)), na.rm = TRUE))
-}
+if (!executeDeconvolution) {
+  # if no signatur mutation found write empty output file
+  # TODO: sombody should check whether this empty file with header is enough, or a more sensible default is required
+  write.table(output_variant_plot, output_variants,
+    sep = "\t",
+    na = "NA", row.names = FALSE, quote = FALSE
+  )
+} else {
+  # get all possible variants
+  all_variants <- colnames(msig_simple[, -which(names(msig_simple) %in% "muts")])
+  # add columns for all possible variants to the dataframe
+  for (variant in all_variants) {
+    output_variant_plot[,variant] <- numeric()
+  }
+  meta_data <- c (samplename = sample_name,
+                  dates = date,
+                  location_name = location_name,
+                  coordinates_lat = coordinates_lat,
+                  coordinates_long = coordinates_long)
+
+  output_variant_plot <- bind_rows(output_variant_plot, meta_data)
+
+  # get rownumber for current sample
+  sample_row <- which(grepl(sample_name, output_variant_plot$samplename))
+
+  # write mutation frequency values to df
+  for (i in all_variants){
+    if (i %in% df$name){
+      # check if variant already has a column
+      if (i %in% colnames(output_variant_plot)) {
+          output_variant_plot[sample_row,][i] <- df$value[df$name == i]
+          output_variant_plot <- output_variant_plot %>% mutate(others = 1-rowSums(across(all_of(all_variants)), na.rm = TRUE))
+  }
+    }
   }
-}
 
-# 2. check if file exists already
-if (file.exists (output_variants)) {
-  previous_df <- read.table (output_variants, 
-                             sep = "\t", header = TRUE, colClasses = "character", check.names = FALSE)
-  # convert numeric values to character
-  output_variant_plot <- as.data.frame(lapply(output_variant_plot, as.character), check.names = FALSE)
-  # merge with adding cols and rows
-  output_variant_plot <- full_join(previous_df, output_variant_plot, by = colnames(previous_df), copy = TRUE)
+  ## # TODO: This chunk hast to go into a seperate rule
```

Comment:
Let's delete this instead of commenting it.  It's either needed or not.  Let's not clutter this already cluttered file :)